src/densities.jl

```diff
@@ -79,7 +80,10 @@ is not collinear the spin density is `nothing`.
 
     Threads.@threads for (ikpts, ρaccu) in collect(zip(kpt_per_thread, ρaccus))
         ρaccu .= 0
-        ρ_k = similar(ψ[1][:, 1], basis.fft_size)
+
+        # TODO use something like T = promote_type(Array{eltype(basis)}, eltype(ψ), ...)
+        ρ_k = similar(Array{complex(T)}, basis.fft_size)
```

Comment:
this allocates twice

Reply:
Why twice? This just allocates the array and that's it.

Reply:
oh right, sorry, not used to this pattern